src/iris/nodes/segmentation/tensorrt_multilabel_segmentation.py

```diff
@@ -105,15 +109,26 @@ def __init__(
             input_num_channels (Literal[1, 3]): Model input image number of channels. Defaults to 3.
             callbacks (List[Callback], optional): List of algorithm callbacks. Defaults to [].
         """
-        engine = self._load_engine(model_path)
-
-        segmap_output_shape = engine.get_binding_shape(1)
-        inputs, outputs, bindings, stream = self._allocate_buffers(engine)
-        context = engine.create_execution_context()
+        self.engine = self._load_engine(model_path)
+        self.segmap_input_tensor_name = segmap_input_tensor_name
```

Comment:
No need to create self variable here because it's an unnecessary copy. `engine` will be stored in `params` variable after `super().__init__` is called. Also `segmap_input_tensor_name` should be added to `Parameters` class to keep code base consistency.
```suggestion
        engine = self._load_engine(model_path)
```

Reply:
Done.